### PR TITLE
Modify RunInference to return PipelineResult for the benchmark tests

### PR DESCRIPTION
**Pull request by AnandInguva**
RunInference benchmarks tests uses the [example](https://github.com/apache/beam/tree/master/sdks/python/apache_beam/examples/inference) to run the pipeline. 
 * These tests utilizes the `LoadTest` class to publish metrics to BigQuery and InfluxDB. 

This PR modifies the example pipeline to use a `test_pipeline` param passed from the Inference LoadTests/benchmark tests

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by AnandInguva** [comment]
Run Python 3.7 PostCommit

**Comment by AnandInguva** [comment]
> Where/how will the `PipelineResult` be used? do you have an example?
> 
> These tests are skipped for now in the PostCommits. Did you test locally?

Yes, i tested them and they run locally

**Comment by AnandInguva** [comment]
https://ci-beam.apache.org/job/beam_PostCommit_Python37_PR/361/console 

I un-skipped the tests and ran it on the Jenkins as well. This failed due to a different test but the InferenceTests pass here

**Review by yeandy — APPROVED** [review]

**Comment by AnandInguva** [comment]
R: @tvalentyn . This PR is needed for the benchmark tests as I need `PipelineResult` object to publish metrics to the BQ using `LoadTest` class

File: sdks/python/apache_beam/examples/inference/pytorch_image_classification.py [hunk 1]
```diff
@@ -95,21 +98,28 @@ def parse_known_args(argv):
...
+    save_main_session=True,
+    test_pipeline=None) -> PipelineResult:
   """
   Args:
     argv: Command line arguments defined for this example.
     model_class: Reference to the class definition of the model.
     model_params: Parameters passed to the constructor of the model_class.
                   These will be used to instantiate the model object in the
                   RunInference API.
+    test_pipeline: used for internal testing. No backwards-compatibility,
```
[the remark this answers is not in the document]
Comment (tvalentyn) [reply sdks/python/apache_beam/examples/inference/pytorch_image_classification.py:114]: Add that save_main_session is used for testing only as well.

File: sdks/python/apache_beam/examples/inference/pytorch_image_classification.py [hunk 2]
```diff
@@ -120,27 +130,35 @@ def run(argv=None, model_class=None, model_params=None, save_main_session=True):
...
-        filename_value_pair
-        | 'PyTorchRunInference' >> RunInference(model_handler)
-        | 'ProcessOutput' >> beam.ParDo(PostProcessor()))
-
-    if known_args.output:
-      predictions | "WriteOutputToGCS" >> beam.io.WriteToText( # pylint: disable=expression-not-assigned
-        known_args.output,
-        shard_name_template='',
-        append_trailing_newlines=True)
+  if not test_pipeline:
```
Comment (tvalentyn) [inline sdks/python/apache_beam/examples/inference/pytorch_image_classification.py:133]: you could s/`test_pipeline`/`pipeline` in args, and delete 135-136.

File: sdks/python/apache_beam/examples/inference/pytorch_image_classification.py [hunk 3]
```diff
@@ -72,6 +74,7 @@ def parse_known_args(argv):
   """Parses args for the workflow."""
   parser = argparse.ArgumentParser()
   parser.add_argument(
+      '--input_file',
```
[the remark this answers is not in the document]
Comment (AnandInguva) [reply sdks/python/apache_beam/examples/inference/pytorch_image_classification.py:77]: Removed input_file.

File: sdks/python/apache_beam/examples/inference/pytorch_image_classification.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (AnandInguva) [reply sdks/python/apache_beam/examples/inference/pytorch_image_classification.py:114]: Done

File: sdks/python/apache_beam/examples/inference/pytorch_image_classification.py (hunk 2 above)
Comment (AnandInguva) [reply sdks/python/apache_beam/examples/inference/pytorch_image_classification.py:133]: This test_pipeline comes from the `load_test` and after the pipeline is ran, the test_pipeline object is used in the `load_test` to publish metrics. So I need its state to be preserved through out the E2E test. That's why I passed it as an argument to the example.

**Comment by AnandInguva** [comment]
Run Python 3.7 PostCommit

**Comment by AnandInguva** [comment]
Run Python 3.9 PostCommit

File: sdks/python/apache_beam/examples/inference/pytorch_image_classification.py (hunk 2 above)
Comment (tvalentyn) [reply sdks/python/apache_beam/examples/inference/pytorch_image_classification.py:134]: I understand that, i meant that we could call the `test_pipeline` args just `pipeline`, and then have the code like:

```
...
    pipeline: used for internal testing. No backwards-compatibility,
...
if not pipeline:
   pipeline = ...
```

**Comment by tvalentyn** [comment]
Run Python Precommit

**Comment by AnandInguva** [comment]
Python 3.7 PostCommit errors are not related to this change.

File: sdks/python/apache_beam/examples/inference/sklearn_mnist_classification.py [hunk 4]
```diff
@@ -60,7 +62,7 @@ def parse_known_args(argv):
   """Parses args for the workflow."""
   parser = argparse.ArgumentParser()
   parser.add_argument(
-      '--input_file',
+      '--input',
```
Comment (tvalentyn) [inline sdks/python/apache_beam/examples/inference/sklearn_mnist_classification.py:65]: let's update the markdown at: https://github.com/apache/beam/tree/master/sdks/python/apache_beam/examples/inference

to reflect this change. can be a separate pr. 
@rszper , @rezarokni are there any documentations/notebooks where we provide complete command line to run examples and might need to fix this?

Thanks.

Comment (rszper) [reply sdks/python/apache_beam/examples/inference/sklearn_mnist_classification.py:65]: Not in the documentation that I created. @AnandInguva has been working on notebooks, but I'm not sure if they would be related to this change.

Comment (tvalentyn) [reply sdks/python/apache_beam/examples/inference/sklearn_mnist_classification.py:65]: thanks!

Comment (AnandInguva) [reply sdks/python/apache_beam/examples/inference/sklearn_mnist_classification.py:65]: +1 with @rszper. The markdown seems fine with the current implementation as well.

**Comment by AnandInguva** [comment]
Changed md file to reflect the change. Thanks. PTAL @tvalentyn

**Comment by AnandInguva** [comment]
Run Python 3.7 PostCommit

**Comment by AnandInguva** [comment]
Run Python 3.9 PostCommit

**Comment by AnandInguva** [comment]
Run Python 3.7 PostCommit

**Comment by AnandInguva** [comment]
Run Python 3.9 PostCommit

**Comment by AnandInguva** [comment]
PTAL : @tvalentyn

**Review by tvalentyn — APPROVED** [review]
